ethereum_clients/ControlledProcess.js

```diff
@@ -122,11 +135,13 @@ class ControlledProcess extends EventEmitter {
           let parts = log.split(/\r|\n/)
           parts = parts.filter(p => !['', '> '].includes(p))
           this.logs.push(...parts)
-          parts.map(l => {
-            this.emit('log', l)
-            if (l.toLowerCase().includes('error')) {
-              this.emit('error', l)
-              this.debug('Emit: error', l)
+          parts.map(logPart => {
+            this.emit('log', logPart)
+            if (this.handleData) {
+              this.handleData(logPart, this.emit.bind(this), Notification)
```

Comment:
since handleData has no return or callback this can be implemented in the plugin within a log-event listener?

Reply:
hm yes i can try that, you  mean something like below?

```js
this.on('log', log => {
  if (this.handleData) {
    this.handleData(logPart, this.emit.bind(this), Notification)
  }
})
```